frontend/src/Components/RouteWithAuthentication.tsx

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import { Redirect, Route, RouteComponentProps } from "react-router-dom";
+import { isAuthenticated } from "../Utilities/AuthenticationCookies";
+
+interface IProps {
+  Component: React.ComponentType;
+  onlyLoggedIn: boolean;
+  redirectRoute: string;
+  path: string;
+  exact?: boolean;
+}
+
+const RouteWithAuthentication = ({
+  Component,
+  onlyLoggedIn,
+  redirectRoute,
+  path,
+  exact,
+}: IProps) => {
+
+  const render = (props: any) =>
+    isAuthenticated() === onlyLoggedIn ? (
```

Comment:
`Lambdas are forbidden in JSX attributes due to their rendering performance impact _tslint(jsx-no-lambda)_`